.github/workflows/precommit.yaml

```diff
@@ -0,0 +1,23 @@
+---
+name: Pre-commit
+"on":
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Install Python dependencies
+        run: pip install -r requirements.txt
```

Comment:
did you check if there is a container we can use with Python and pre-commit already installed, instead of installing them on the ubuntu machine for each run?

Reply:
I didn't find any official container recommended for this use, only random docker-hub ones. I did see a lot of repositories using the same method of installing on a fresh ubuntu machine.